### PR TITLE
Feature x26 password reset email

### PR DESCRIPTION
**Pull request by MykhailoGrynenko**
Password reset functionality via email has been added.

### CODE REVIEW

**Review by OlehHnyp — COMMENTED** [review]
I tried to reset password and received email with link to "Example Domain".
I didn't succeed to reset password.  
How about asking more information(phone number, first name, last name)  before sending password-reset email?

File: ProjectShop/UserApp/urls.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from UserApp.views import LoginView, RegisterView
+
+urlpatterns = [
+    path('login/', LoginView.as_view(), name='login'),
+    path('register/', RegisterView.as_view(), name='register'),
+    path('logout/', auth_views.LogoutView.as_view(template_name='UserApp/logout.html'), name='logout'),
+    path('password-reset/',
+         auth_views.PasswordResetView.as_view(template_name='UserApp/password_reset.html'),
+         name='password_reset'),
```
[the remark this answers is not in the document]
Comment (MykhailoGrynenko) [reply ProjectShop/UserApp/urls.py]: Дякую. Зробив зміни. Прошу перевірити.

File: ProjectShop/UserApp/views.py [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+class LoginView(auth_views.LoginView):
+    form_class = LoginForm
+    template_name = 'UserApp/login.html'
+
+
+class RegisterView(SuccessMessageMixin, generic.CreateView):
+    form_class = RegisterForm
+    template_name = 'UserApp/register.html'
+    success_url = reverse_lazy('login')
+    success_message = "Congratulations! Your account has been created. You may sign in!"
```
[the remark this answers is not in the document]
Comment (MykhailoGrynenko) [reply ProjectShop/UserApp/views.py:19]: Дякую.
Начебто достатньо цього:
from django.utils.translation import ugettext_lazy as _
success_message = _("Congratulations! Your account has been created. You may sign in!")
Screenshot from 2021-09-17 15-02-56

Чи краще зробити через метод get_success_message?

[the remark this answers is not in the document]
Comment (strangerok1507) [reply ProjectShop/UserApp/views.py:19]: success_message = _("Congratulations! Your account has been created. You may sign in!")
всё супер)

**Review by strangerok1507 — CHANGES_REQUESTED** [review]

**Review by strangerok1507 — APPROVED** [review]
LGTM +1

**Comment by v-lavrushko** [comment]
closes #23 
closes #26

**Pull request merged**